### PR TITLE
Truncate columns list to match tr_frame for correct dict formatters lookup

### PR DESCRIPTION
**Pull request by kesmit13**
When using a dictionary of formatters and a truncated DataFrame is being displayed, the column list used for looking up the formatter in the dict are incorrect.  The columns attribute of the DataFrameFormatter needs to be truncated as well for the lookup to be correct.

- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by WillAyd** [comment]
I'm not sure that overwriting self.columns is the way to go here - is there a way to localize the fix?

**Comment by kesmit13** [comment]
I think the new pull request localizes it down to just the spot where the fix is needed.  My original fix was based on a monkey patch I was using in my library to make it work in older versions of pandas.

File: pandas/tests/io/formats/test_format.py
Comment (WillAyd) [inline pandas/tests/io/formats/test_format.py:707]: Rather than this can you just parametrize the inputs?

**Review by WillAyd — CHANGES_REQUESTED** [review]
Cool I think this is a better approach. Can you add a whatsnew note for 1.2?

File: pandas/tests/io/formats/test_format.py
Comment (kesmit13) [reply pandas/tests/io/formats/test_format.py:707]: I'm not sure what you mean.

Comment (WillAyd) [reply pandas/tests/io/formats/test_format.py:707]: Check out the `pytest.mark.parametrize` decorator - it is used by a few other tests in this module already

Comment (kesmit13) [reply pandas/tests/io/formats/test_format.py:707]: I haven't used that before, but I'm still not sure what you had in mind to parameterize this method.

Comment (kesmit13) [reply pandas/tests/io/formats/test_format.py:714]: I added parameterized inputs.  I'm not sure if this is quite what you were expecting since there was only one pair of inputs / outputs used in the test.

Comment (ivanovmg) [reply pandas/tests/io/formats/test_format.py:714]: I would argue that the original version (without parametrize) was more readable. Maybe @WillAyd suggested that you parameterize separately formatters for float, int and object?

Comment (kesmit13) [reply pandas/tests/io/formats/test_format.py:714]: That wouldn't exercise the fix though.  It needs all of the columns in one DataFrame so that a truncated string representation can be displayed.

Comment (WillAyd) [reply pandas/tests/io/formats/test_format.py:714]: The suggestion was to parametrize the arguments to `formatters`; one parametrize for the dict input and one for the list

**Comment by kesmit13** [comment]
AFAIK, I've addressed all of the issues. I'm just waiting on final approval.

**Review by arw2019 — COMMENTED** [review]
@kesmit13 Can you merge master again and resolve conflicts?

FWIW lgtm

**Comment by jreback** [comment]
@kesmit13 yeah can you merge master and fix conflict.

**Comment by ivanovmg** [comment]
After recent changes in ``pandas/io/format/format.py`` it will be necessary to move changes in ``_get_formatter`` method into the same method in class ``DataFrameFormatter``.
And you probably won't need to type ``tr_frame``.

File: doc/source/whatsnew/v1.2.0.rst
Comment (ivanovmg) [inline doc/source/whatsnew/v1.2.0.rst:468]: Currently there is no ``TableFormatter``.
The generic functionality is moved to ``DataFrameFormatter``.

**Review by ivanovmg — CHANGES_REQUESTED** [review]

File: pandas/tests/io/formats/test_to_string.py
Comment (ivanovmg) [inline pandas/tests/io/formats/test_to_string.py:176]: Can you please fix test?
Apparently, ``self`` should be removed here.

**Comment by ivanovmg** [comment]
@kesmit13, can you please merge master and resolve conflicts in ``doc/source/whatsnew/v1.2.0.rst``?
Without it we cannot witness CI checks.
Thank you!

**Review by ivanovmg — COMMENTED** [review]
Sorry for a really late reply but can you please merge master again?
I guess that your whatsnew entry should go now to v1.3.

**Comment by simonjayhawkins** [comment]
> I guess that your whatsnew entry should go now to v1.3.

@kesmit13 can you address @ivanovmg comment

**Comment by jreback** [comment]
can you merge master and will have a look (also note would need to be in 1.4)

**Comment by mroeschke** [comment]
Thanks for the PR, but it appears to have gone stale. This appears close, so if you could merge master and address the remaining comments, we'd be happy to reopen and have another look over.

**Pull request closed without merging**
